### PR TITLE
src: implement `v8 inspect --array-length` for large arrays

### PR DESCRIPTION
**Pull request by joyeecheung**
Also added tests for `--string-length` since they are pretty similar...

Refs: https://github.com/nodejs/llnode/issues/89

cc:/ @bnoordhuis

### CODE REVIEW

File: src/llv8.cc [hunk 1]
```diff
@@ -1838,10 +1838,10 @@ v8::Value JSObject::GetArrayElement(int64_t pos, Error& err) {
 }
 
 std::string JSArray::Inspect(InspectOptions* options, Error& err) {
-  Smi length = Length(err);
+  int64_t length = GetArrayLength(err);
```
Comment (joyeecheung) [inline src/llv8.cc]: The original code would call `NameDictionary::Length()` instead so the length displayed would count an additional entry (`length`).

File: src/llv8.cc [hunk 2]
```diff
@@ -1838,12 +1847,16 @@ v8::Value JSObject::GetArrayElement(int64_t pos, Error& err) {
...
   if (err.Fail()) return std::string();
 
-  std::string res = "<Array: length=" + length.ToString(err);
+  std::string res = "<Array: length=" + std::to_string(length);
   if (options->detailed) {
-    std::string elems = InspectElements(err);
+    int64_t display_length = length;
+    if (length > options->array_length) {
+      display_length = options->array_length;
+    }
```
Comment (bnoordhuis) [inline src/llv8.cc]: `int64_t display_length = std::min(length, options->array_length);`?  Arguably a little easier to read.

If not that, can you `s/length/display_length/` in the comparison?

File: test/inspect-test.js [hunk 3]
```diff
@@ -72,6 +74,16 @@ tape('v8 inspect', (t) => {
     t.ok(/.other-key=[^\n]*<String: "ohai">/.test(lines),
          '.other-key property');
 
+    const arrayMatch = lines.match(
+    /.array=(0x[0-9a-f]+):<Array: length=6>/);
```
Comment (bnoordhuis) [inline test/inspect-test.js]: Can you indent by 4 spaces here and elsewhere?  (Better, indent but break after the `=`.)

**Comment by joyeecheung** [comment]
@bnoordhuis @indutny  Thanks for the review, updated per https://github.com/nodejs/llnode/pull/105#pullrequestreview-45466113 PTAL.

I have tested this with v4.8.3, v6.10.0 and v8.0.0 on MacOS 10.12.4, all tests have passed.

File: test/inspect-test.js [hunk 4]
```diff
@@ -72,6 +74,16 @@ tape('v8 inspect', (t) => {
     t.ok(/.other-key=[^\n]*<String: "ohai">/.test(lines),
          '.other-key property');
 
+    const arrayMatch = 
+      lines.match(/.array=(0x[0-9a-f]+):<Array: length=6>/);
```
Comment (bnoordhuis) [inline test/inspect-test.js]: Can you use 4 space indent here and below?

**Comment by joyeecheung** [comment]
@bnoordhuis Fixed the indentation, PTAL.

File: test/inspect-test.js [hunk 5]
```diff
@@ -96,6 +108,40 @@ tape('v8 inspect', (t) => {
...
+  });
+
+  sess.linesUntil(/}>/, (lines) => {
+    lines = lines.join('\n');
+    t.notEqual(
+        lines.indexOf('<Array: length=6'),
+        -1,
+        'array length');
+    t.ok(
+        lines.match(/\[5\]=0x[0-9a-f]+:<function: Class at .+\.js:\d+:\d+>}>$/),
```
Comment (cjihrig) [inline test/inspect-test.js]: It doesn't look like you're using the result of `match()` here and on line 143. It might be better to use `test()` instead.

**Pull request merged**